exercises/concept/windowing-system/.docs/instructions.md

```diff
@@ -0,0 +1,159 @@
+// => 1080
+```
+
+## 2. Define a `position` struct to store a window position
+
+Define a struct (or class) named `position` with two member variables, `x` and `y` that store the current horizontal and vertical position, respectively, of the window's upper left corner.
+The constructor function should accept initial values for these fields.
+The value for `x` is provided as the first parameter, the value for `y` as the second one.
+The default value should be `0` for both variables.
+
+The position (0, 0) is the upper left corner of the screen with `x` values getting larger as you move right and `y` values getting larger as you move down.
+
+Also define a method `void move(int newX, int newY)` that takes new x and y parameters and changes the properties to reflect the new position.
+
+```cpp
+position point{};
+point.x;
+// => 0
+point.y;
+// => 0
+
+point.move(100, 200);
+point.x;
+// => 100
+point.y;
+// => 200
+```
+
+## 3. Define a `programWindow` class
+
+Define a `programWindow` class or struct with __pointers__ to the following member variables:
+
+- `screenSize`: holds a pointer of type `size` with `width` 800 and `height` 600
+- `windowSize` : holds a pointer of type `size`, the initial value is the default value of the `size` instance
+- `windowPosition` : holds a pointer of type `position`, the initial value is the default value of the `position` instance
+
+When the window is opened (initialized), it always has the default `windowSize` and `windowPosition` in the beginning.
+Its constructor should not take any input parameters.
+
+```cpp
+programWindow aProgramWindow{};
+programWindow.screenSize->width;
+// => 800
+
+// Similar for the other fields.
+```
+
+## 4. Add a method to resize the window
+
+The `programWindow` class should include a function `resize`.
+It should accept a __pointer__ of type `size` as input and attempts to resize the window to the specified size.
+
+However, the new size cannot exceed certain bounds.
+
+- The minimum allowed height or width is 1.
+  Requested heights or widths less than 1 will be clipped to 1.
+- The maximum height and width depend on the current position of the window, the edges of the window cannot move past the edges of the screen.
+  Values larger than these bounds will be clipped to the largest size they can take.
+  E.g. if the window's position is at `x` = 400, `y` = 300 and a resize to `height` = 400, `width` = 300 is requested, then the window would be resized to `height` = 300, `width` = 300 as the screen is not large enough in the `y` direction to fully accommodate the request.
+
+```cpp
+programWindow aProgramWindow{};
+
+size newSize{600, 400};
+aProgramWindow.resize(&newSize);
+aProgramWindow.size->width;
+// => 600
+aProgramWindow.size->height;
+// => 400
+```
+
+## 5. Add a method to move the window
+
+Besides the resize functionality, the `programWindow` class should also include a function `move`.
+It should accept a parameter as a __pointer__ of type `position` as input.
```

Comment:
instead of "a parameter ... as input" I'd suggest either "a parameter ..." or "... as input".